src/main/java/com/endava/webhw/controller/DepartmentController.java

```diff
@@ -0,0 +1,56 @@
+package com.endava.webhw.controller;
+
+import com.endava.webhw.dto.DepartmentDto;
+import com.endava.webhw.model.Department;
+import com.endava.webhw.service.DepartmentService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.mvc.method.annotation.MvcUriComponentsBuilder;
+
+import java.net.URI;
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/departments")
+public class DepartmentController {
+    private final DepartmentService departmentService;
+
+    @GetMapping
+    public List<Department> findAll() {
+        return departmentService.findAll();
+    }
+
+    @GetMapping("{id}")
+    public Department findById(@PathVariable long id) {
+        return departmentService.findById(id);
+    }
```

Comment:
I think you need to return a DTO not Entity
Why? Imagine in next commit you update your Entity to contain Department Entity but consumers of your API await for department_id field but after your commit there will be department entity instead so you will force people to update their code to adapt
While having DTO for request and responses you detach that coupling with your mappers DTO->Entity->DTO

Reply:
Changed the Controllers and Services to behave this way in the latest commit.